pkg/api/dashboard_snapshot.go

```diff
@@ -325,22 +330,26 @@ func (hs *HTTPServer) DeleteDashboardSnapshot(c *models.ReqContext) response.Res
 		// all permissions must be explicit, so the lack of a rule for dashboard 0 means the guardian will reject.
 		dashboardID := query.Result.Dashboard.Get("id").MustInt64()
 
-		guardian := guardian.New(c.Req.Context(), dashboardID, c.OrgID, c.SignedInUser)
-		canEdit, err := guardian.CanEdit()
-		// check for permissions only if the dahboard is found
-		if err != nil && !errors.Is(err, dashboards.ErrDashboardNotFound) {
-			return response.Error(500, "Error while checking permissions for snapshot", err)
-		}
+		// If for some reason the snapshot is not external and yet targets no dashboard (ID set 0), RBAC access-control
+		// can be skipped.
+		if dashboardID != 0 {
+			guardian := guardian.New(c.Req.Context(), dashboardID, c.OrgID, c.SignedInUser)
+			canEdit, err := guardian.CanEdit()
+			// check for permissions only if the dashboard is found
+			if err != nil && !errors.Is(err, dashboards.ErrDashboardNotFound) {
+				return response.Error(http.StatusInternalServerError, "Error while checking permissions for snapshot", err)
+			}
 
-		if !canEdit && query.Result.UserId != c.SignedInUser.UserID && !errors.Is(err, dashboards.ErrDashboardNotFound) {
-			return response.Error(403, "Access denied to this snapshot", nil)
+			if !canEdit && query.Result.UserId != c.SignedInUser.UserID && !errors.Is(err, dashboards.ErrDashboardNotFound) {
+				return response.Error(http.StatusForbidden, "Access denied to this snapshot", nil)
+			}
 		}
```

Comment:
If the comment above doesn't lie, you should be able to remove the `else` that is surrounding this code for the same effect, since an external dashboard is also going to return ID 0.